### PR TITLE
Fix the structure of clustering config

### PR DESCRIPTION
**Pull request by TkrUdagawa**
I'm trying to fix the structure of clustering config like:

```
{
 "method": "kmeans",
 "parameter": {
    "k": 3
  },
  "compressor_method" : "simple",
  "compressor_parameter" : {
     "bucket_size": 10
  }
}
```

This PR includes fixing factory methods (clustering factory, clustering method factory and storage factory), removing the dependency of `clustering_config` and fixing related classes. 

I need to improve code quality and solve conflictions.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Also fix https://github.com/jubatus/jubatus/tree/master/config/clustering and client tests.

**Comment by TkrUdagawa** [comment]
I have updated codes to resolve conflictions and cpplint errors.

File: jubatus/core/driver/clustering_test.cpp [hunk 1]
```diff
@@ -47,29 +48,144 @@ using jubatus::util::lang::lexical_cast;
...
+      bucket_size(10000),
+      bucket_length(2),
+      bicriteria_base_size(10),
+      compressed_bucket_size(200),
+      forgetting_factor(2.0),
+      forgetting_threshold(0.05),
+      seed(0),
+      eps(2.0),
+      min_core_point(1) {
+
```
Comment (yukimori) [inline jubatus/core/driver/clustering_test.cpp]: There is blank line at the end of code block.

File: jubatus/core/driver/clustering_test.cpp [hunk 2]
```diff
@@ -47,29 +48,144 @@ using jubatus::util::lang::lexical_cast;
...
+  }
+
+  int k;
+
```
Comment (yukimori) [inline jubatus/core/driver/clustering_test.cpp]: unnecessary black code

**Comment by yukimori** [comment]
LGTM!

**Comment by kmaehashi** [comment]
Refs #335

**Pull request merged**